src/Dialect/ONNX/ElementsAttr/ElementsAttrBuilder.cpp

```diff
@@ -100,6 +102,74 @@ DenseElementsAttr ElementsAttrBuilder::toDenseElementsAttr(
   llvm_unreachable("unexpected ElementsAttr instance");
 }
 
+/*static*/
+bool ElementsAttrBuilder::equal(ElementsAttr lhs, ElementsAttr rhs) {
+  auto lhsType = lhs.getType();
+  auto rhsType = rhs.getType();
+  auto elementType = lhsType.getElementType();
+
+  // Equality between signed/unsigned or int/fp not supported.
+  assert(wideBTypeOfBType(btypeOfMlirType(elementType)) ==
```

Comment:
I believe it would be confusing to work around callers passing incompatible arguments. I added the comment:
```
  // Asserts if these preconditions are violated (doesn't return false as that
  // would hide whether the lhs and rhs are different or incompatible).
```